ttnn/cpp/ttnn/operations/moreh/moreh_norm/moreh_norm.cpp

```diff
@@ -20,25 +22,49 @@ Tensor MorehNorm::invoke(
+    auto INF = std::numeric_limits<float>::infinity();
+    if (auto single_dim = std::get_if<int64_t>(&dim.value())) {
+        if (p == 0.0 || p == INF || p == -INF) {
+            return ttnn::prim::moreh_norm(input, p, *single_dim, keepdim, output, memory_config, compute_kernel_config);
+        }
+        auto tmp_output =
+            ttnn::prim::moreh_norm(input, p, *single_dim, true, std::nullopt, memory_config, compute_kernel_config);
+        return ttnn::moreh_root(tmp_output, p, *single_dim, keepdim, output, memory_config, compute_kernel_config);
+    }
 
     auto dims = std::get<ttnn::SmallVector<int64_t>>(dim.value());
     if (dims.empty()) {
         ttnn::SmallVector<int64_t> all_dims(input.get_legacy_shape().rank());
         std::iota(all_dims.begin(), all_dims.end(), 0);
         dims = all_dims;
     }
-    if (dims.size() == 1)
-        return ttnn::prim::moreh_norm(input, p, dims[0], keepdim, output, memory_config, compute_kernel_config);
-
+    if (dims.size() == 1) {
+        if (p == 0.0 || p == INF || p == -INF) {
+            return ttnn::prim::moreh_norm(input, p, dims[0], keepdim, output, memory_config, compute_kernel_config);
+        }
+        auto tmp_output =
+            ttnn::prim::moreh_norm(input, p, dims[0], true, std::nullopt, memory_config, compute_kernel_config);
+        return ttnn::moreh_root(tmp_output, p, dims[0], keepdim, output, memory_config, compute_kernel_config);
+    }
     std::sort(dims.begin(), dims.end(), std::greater<int64_t>());
     auto tmp_output =
         ttnn::prim::moreh_norm(input, p, dims.front(), keepdim, std::nullopt, memory_config, compute_kernel_config);
     using idx_t = decltype(dims.size());
-    for (idx_t idx = 1; idx < dims.size() - 1; ++idx)
-        tmp_output = ttnn::prim::moreh_norm(
-            tmp_output, p, dims[idx], keepdim, std::nullopt, memory_config, compute_kernel_config);
-    return ttnn::prim::moreh_norm(tmp_output, p, dims.back(), keepdim, output, memory_config, compute_kernel_config);
+    if (p == 0.0) {
+        dims.erase(dims.begin());
+        return ttnn::moreh_sum(tmp_output, dims, keepdim, output, memory_config, compute_kernel_config);
+    } else if (p == INF || p == -INF) {
+        using idx_t = decltype(dims.size());
+        for (idx_t idx = 1; idx < dims.size() - 1; ++idx) {
+            tmp_output = ttnn::prim::moreh_norm(
+                tmp_output, p, dims[idx], keepdim, std::nullopt, memory_config, compute_kernel_config);
+        }
+        return ttnn::prim::moreh_norm(
+            tmp_output, p, dims.back(), keepdim, output, memory_config, compute_kernel_config);
+    } else {
+        dims.erase(dims.begin());
+        tmp_output = ttnn::moreh_sum(tmp_output, dims, true, std::nullopt, memory_config, compute_kernel_config);
+        return ttnn::moreh_root(tmp_output, p, dims.back(), keepdim, output, memory_config, compute_kernel_config);
+    }
```

Comment:
It seems that this change might affect performance and memory usage.

Reply:
I expected the performance to drop because of the multiple ops called but I'm not sure there is a better way currently

Reply:
> I expected the performance to drop because of the multiple ops called but I'm not sure there is a better way currently

I think it would be easier to understand if you separate it into forms like MorehNorm(where p is a number), MorehNormZero, and MorehNormInf(where p is infinity or negative infinity).
Since MorehNorm(where p is a number) is already implemented, there's no need to modify it.

Reply:
Please rename the function to represent appropriate behavior.
Moreover, is it available to fuse the existing prim ops so that the norm op consists of minimum number of programs?

Reply:
I figured I can reduce the number of prim ops need to called down to 2 (using an ops similar to `torch.sum` and `moreh_sum`) but it would still need at least 3 program calls.

Reply:
OK.
Please make an op consists of minimum number of prim ops for the performance.